src/uploaders.ts

```diff
@@ -33,26 +35,38 @@ export class XHRUploader extends AsyncUploader {
     public async upload(fetchPayload: fetchPayload): Promise<Response> {
         const response: Response = await this.makeRequest(
             this.url,
-            fetchPayload.body
+            fetchPayload.body,
+            fetchPayload.method as HTTPMethod,
+            fetchPayload.headers
         );
         return response;
     }
 
-    private async makeRequest(url: string, data: string): Promise<Response> {
+    private async makeRequest(
+        url: string,
+        data: string,
+        method: HTTPMethod = 'post',
+        headers?: Record<string, string>
```

Comment:
Should we default this to an empty object since you are doing an `Object.keys` on line 66 below?

```suggestion
        headers?: Record<string, string> = {}
```